clients/include/testing_axpy.hpp

```diff
@@ -17,6 +17,15 @@
 
 using namespace std;
 
+#define CLEANUP()                                                  \
```

Comment:
A different PR could handle cleaning up the rest of the codebase.  For this PR, you could define the unique_ptr< hipmemory* > and not create more code to clean up later.